tools/data/omnisource/preparing_omnisource.md

```diff
@@ -0,0 +1,144 @@
+| k200_raw_train  |  76027   | 963.5G  | SlowOnly-8x8-R50 |           N/A            |             N/A              |
+
+The file structure of our uploaded OmniSource dataset looks like:
+
+```
+OmniSource/
+├── annotations
+│   ├── googleimage_200
+│   │   ├── googleimage_200.txt							File list of all valid images crawled from Google.
+│   │   ├── tsn_8seg_googleimage_200_duplicate.txt		Postive file list of images crawled from Google, which is similar to a validation example.
+│   │   ├── tsn_8seg_googleimage_200.txt				Postive file list of images crawled from Google, filtered by the teacher model.
+│   │   └── tsn_8seg_googleimage_200_wodup.txt			Postive file list of images crawled from Google, filtered by the teacher model, after de-duplication.
+│   ├── insimage_200
+│   │   ├── insimage_200.txt
+│   │   ├── tsn_8seg_insimage_200_duplicate.txt
+│   │   ├── tsn_8seg_insimage_200.txt
+│   │   └── tsn_8seg_insimage_200_wodup.txt
+│   ├── insvideo_200
+│   │   ├── insvideo_200.txt
+│   │   ├── slowonly_8x8_insvideo_200_duplicate.txt
+│   │   ├── slowonly_8x8_insvideo_200.txt
+│   │   └── slowonly_8x8_insvideo_200_wodup.txt
+│   ├── k200_actions.txt								The list of action names of the 200 classes in MiniKinetics.
+│   ├── K400_to_MiniKinetics_classidx_mapping.json		The index mapping from Kinetics-400 to MiniKinetics.
+│   ├── kinetics_200
+│   │   ├── k200_train.txt
+│   │   └── k200_val.txt
+│   ├── kinetics_raw_200
+│   │   └── slowonly_8x8_kinetics_raw_200.json			Kinetics Raw Clips filtered by the teacher model.
+│   └── webimage_200
+│       └── tsn_8seg_webimage_200_wodup.txt				The union of `tsn_8seg_googleimage_200_wodup.txt` and `tsn_8seg_insimage_200_wodup.txt`
+├── googleimage_200										(10 volumes)
+│   ├── vol_0.tar
+│   ├── ...
+│   └── vol_9.tar
+├── insimage_200										(10 volumes)
+│   ├── vol_0.tar
+│   ├── ...
+│   └── vol_9.tar
+├── insvideo_200										(20 volumes)
+│   ├── vol_00.tar
+│   ├── ...
+│   └── vol_19.tar
+├── kinetics_200_train
+│   └── kinetics_200_train.tar
+├── kinetics_200_val
+│   └── kinetics_200_val.tar
+└── kinetics_raw_200_train								(16 volumes)
+    ├── vol_0.tar
+    ├── ...
+    └── vol_15.tar
+```
+
+## Data Preparation
+
+For data preparation, you need to first download those data. For `kinetics_200` and 3 web datasets: `googleimage_200`, `insimage_200` and `insvideo_200`, you just need to extract each volume and merge their contents. For Kinetics raw videos, since loading long videos is very heavy, you need to first trim it into clips. Here we provide a script named `trim_raw_video.py`. It trims a long video into 10-second clips and remove the original raw video. You can use it to trim the Kinetics raw video. The data should be placed in `data/OmniSource/`. When data preparation finished, the folder structure of `data/OmniSource` looks like (We omit the files not needed in training & testing for simplicity):
+
+```
```

Comment:
better to align the explanation at the right from the same beginning.

Reply:
Not quite sure what's the meaning. Now I split the long line into 3 paragraphs

Reply:
It is good now